packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -262,8 +262,12 @@ export class OpenFileDialog extends FileDialog<MaybeArray<FileStatNode>> {
     }
 
     protected accept(): void {
-        if (this.props.canSelectFolders === false && !Array.isArray(this.value)) {
```

Comment:
shortest negative check is `!this.props.canSelectFolders`

Reply:
`selection.fileStat.isDirectory` +1

What about the case of `canSelectFolders` is false and `canSelectMany` is true? Looks like if a user select several folders and files then folders will get through.

It would be nice to create a sample VS Code extension with several commands running different combination of these properties and then run it in VS Code and Theia and see what is the difference. It also would help with testing i had to modify code just to get an idea what goes wrong.

Reply:
>What about the case of canSelectFolders is false and canSelectMany is true?

It hasn't been implemented, but was trying to workaround this feature and temporary, I inverted condition for option `canSelectFiles` and have got this:

![image](https://user-images.githubusercontent.com/1968177/55406448-d0d19f00-5564-11e9-922c-f73bc2960ae6.png)

Looks like we can open folders, but can't select them. I can include this one in PR if it needed. In correct way, adding additional checking if we have to take care ability of folders selection.

Reply:
At this moment, when we pass `false` to `canSelectFolders`, we can get it in result object, when dialog closes.